### PR TITLE
RIA-6864: Created EmailNotificationGenerator to accept personalisatio…

### PR DESCRIPTION
**Pull request by neha-aggarw**
…n as a Map<String, Object>

- Upgraded gov.notify to 3.19.
- Added interfaces for LRPersonalisation to use the new Personalisation map.
- Added DocumentDownloadClient to download a document from dm-store.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

File: charts/ia-case-notifications-api/values.preview.template.yaml [hunk 1]
```diff
@@ -6,5 +6,7 @@ java:
     IA_AIP_FRONTEND_URL: "https://immigration-appeal.{{ .Values.global.environment }}.platform.hmcts.net/"
     IA_EXUI_FRONTEND_URL: "https://manage-case.{{ .Values.global.environment }}.platform.hmcts.net/"
     OPEN_ID_IDAM_URL: "https://idam-web-public.{{ .Values.global.environment }}.platform.hmcts.net"
+    CCD_GW_URL: "http://dm-store-aat.service.core-compute-aat.internal"
```
[the remark this answers is not in the document]
Comment (colmlynch60) [reply charts/ia-case-notifications-api/values.preview.template.yaml]: @rajeshthuraiyur  removed the preview var as it was not required and DM_URL exists in values.yaml

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by hussain-faruk — APPROVED** [review]

**Pull request merged**